### PR TITLE
feat: wait for mined tx before showing success

### PR DESCRIPTION
**Pull request by petia-latinova**

### CODE REVIEW

File: app/src/pages/Campaign.vue [hunk 1]
```diff
@@ -143,9 +144,19 @@ export default {
...
     onDonateConfirm(value) {
+      const donation = value;
       this.loading = true;
       this.donate({ id: this.campaignId, value })
-        .then(() => this.$buefy.toast.open(`Your donation of ${value} was successfuly completed`))
+        .then((tx) => {
+          this.$buefy.toast.open(
+            `Your donation of ${value} was send. Please, wait for confirmation message...`
+          );
+          getProvider()
```
Comment (imollov) [inline app/src/pages/Campaign.vue]: Isn't there wait method on the tx object itself?

File: app/src/pages/Campaign.vue [hunk 2]
```diff
@@ -143,9 +144,19 @@ export default {
       });
     },
     onDonateConfirm(value) {
+      const donation = value;
```
Comment (imollov) [inline app/src/pages/Campaign.vue]: Is this necessary?

File: app/src/pages/Campaign.vue [hunk 3]
```diff
@@ -143,9 +144,19 @@ export default {
...
       this.loading = true;
       this.donate({ id: this.campaignId, value })
-        .then(() => this.$buefy.toast.open(`Your donation of ${value} was successfuly completed`))
+        .then((tx) => {
+          this.$buefy.toast.open(
+            `Your donation of ${value} was send. Please, wait for confirmation message...`
```
Comment (imollov) [inline app/src/pages/Campaign.vue]:
```suggestion
            `Your donation of ${value} was sent. Please, wait for confirmation message...`
```

File: app/src/store/modules/campaigns.js [hunk 4]
```diff
@@ -129,7 +129,7 @@ export default {
     },
     async donate(_, { id, value }) {
       const userWallet = getWallet();
-      await userWallet.sendTransaction({
+      return await userWallet.sendTransaction({
```
Comment (imollov) [inline app/src/store/modules/campaigns.js]: `await` is redundant here, just return the promise:

```suggestion
      return userWallet.sendTransaction({
```

File: app/src/pages/Campaign.vue (hunk 2 above)
Comment (petia-latinova) [reply app/src/pages/Campaign.vue]: Actually there is no need from that.

**Review by imollov — APPROVED** [review]

**Pull request merged**